elastic/kibana
# [Index management] Data stream edit data retention

Fixes: https://github.com/elastic/kibana/issues/165136

## Summary
This PR adds support for editing data retention for a data stream.


##### How to test
1. Start elasticsearch with `yarn es snapshot --license=trial` and serverless search solution with `yarn start`
2. Navigate to console and execute the commands listed below in order to create a policy
3. Navigate to `index management` and click on `data streams` tab
4. Verify that:
* You can edit the data retention, and its reflected in the table and details flyout
* If you set infinite data retention, it should be reflected in the table and details flyout. And when the edit form is open again the toggle should be enabled.
* If infinite toggle is enabled, you shouldnt be allowed to edit data retention values
* If the data stream is managed by ILM, the edit data retention option shouldnt be shown

<details>
<summary>Creating a test data stream</summary>

```
PUT _index_template/b-index-template
{
  "index_patterns": ["b-data-stream*"],
  "data_stream": { },
  "priority": 500,
  "template": {
    "lifecycle": {
      "enabled": false
    }
  },
  "_meta": {
    "description": "Template with data stream lifecycle"
  }
}

PUT _data_stream/b-data-stream

PUT b-data-stream/_bulk
{ "create":{ } }
{ "@timestamp": "2099-05-06T16:21:15.000Z", "message": "192.0.2.42 - - [06/May/2099:16:21:15 +0000] \"GET /images/bg.jpg HTTP/1.0\" 200 24736" }
{ "create":{ } }
{ "@timestamp": "2099-05-06T16:25:42.000Z", "message": "192.0.2.255 - - [06/May/2099:16:25:42 +0000] \"GET /favicon.ico HTTP/1.0\" 200 3638" }
```

</details>


<details>
<summary>Screenshots</summary>

![Screenshot 2023-09-29 at 11 15 50](https://github.com/elastic/kibana/assets/1191206/4c19ac4a-ea5a-4e9a-978f-b1eb29516a97)


</details>
